web-animations/animation-timeline/animation-timeline-play.html

```diff
@@ -0,0 +1,210 @@
+            'AnimationPlayer should be set to the source ' + type(node) + ' object');
+    });
+}, 'Test AnimationTimeline.play(source) method associate the AnimationNode source ' +
+    'with the returned player');
+
+test(function() {
+    var test = this;
+    var parents = [AnimationGroup, AnimationSequence];
+    parents.forEach(function(parentCtor) {
+        var nodes = [
+            new Animation(createDiv(test), KEYFRAMES, ANIMATION_DURATION),
+            new AnimationGroup([]),
+            new AnimationSequence([])
+        ];
+        nodes.forEach(function(node) {
+            var parent = new parentCtor([node]);
+            var player = document.timeline.play(node);
+
+            assert_equals(player.source, node, 'The source attribute of the created ' +
+                'AnimationPlayer should be set to source parameter ' + type(node) + ' object');
+            assert_equals(node.parent, null, type(node) + ' object, specified ' +
+                'by source parameter should be removed from its parent');
+            assert_array_equals(parent.children, [], type(node) + ' object, specified ' +
+                'by source parameter should be removed from its parent');
+        });
+    });
+}, 'Test AnimationTimeline.play(source) method remove AnimationNode source from its parent');
+
+test(function() {
+    var animationNodes = [
+        new Animation(createDiv(this), KEYFRAMES, ANIMATION_DURATION),
+        new AnimationGroup([]),
+        new AnimationSequence([])
+    ];
+    animationNodes.forEach(function(node, index) {
+        var player = document.timeline.play(node);
+        assert_equals(player.timeline, document.timeline, 'Node ' + index + ': AnimationPlayer, ' +
+            'created by play() method should be accociated with this timeline');
+    });
+}, 'Test AnimationTimeline.play() method accociates returned AnimationPlayer ' +
+    'with timeline instance used to call method play()');
+
+test(function() {
+    var doc = newHTMLDocument();
+    var animationNodes = [
+        new Animation(createDiv(this, doc), KEYFRAMES, ANIMATION_DURATION),
+        new AnimationGroup([]),
+        new AnimationSequence([])
+    ];
+    animationNodes.forEach(function(node, index) {
+        var player = doc.timeline.play(node);
+        assert_equals(player.timeline, doc.timeline, 'Node ' + index +
+            ': AnimationTimeline.play() should associate AnimationPlayer with this timeline');
+    });
+}, 'Test AnimationTimeline.play() method accociates returned AnimationPlayer ' +
+    'with timeline instance used to call method play(). Using timeline of another document');
+
+async_test(function(t) {
+    var player = document.timeline.play(new Animation(createDiv(t), KEYFRAMES, ANIMATION_DURATION));
+
+    player.ready.then(t.step_func(function() {
+        setTimeout(t.step_func(function() {
+            assert_approx_equals(player.currentTime, 50, TIME_EPSILON, 'AnimationTimeline.play() ' +
```

Comment:
Mixing the ANIMATION_DURATION constant with magic numbers like 50 forces the reader to look up ANIMATION_DURATION in testcommon.js. The use of magic numbers or constants should be consistent without mixing them together.
Please resolve this here and elsewhere in this change.